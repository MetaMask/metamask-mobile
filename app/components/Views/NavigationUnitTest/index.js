/**
 * This view was created in order to test the navigation api since it's possible it can change even with minor upgrades.
 * For reference see: https://reactnavigation.org/docs/navigation-prop/#dangerouslygetstate
 */

/* eslint-disable react/prop-types */
import React from 'react';
import { createStackNavigator } from '@react-navigation/stack';
import {
  NavigationContainer,
  useNavigationState,
} from '@react-navigation/native';
import { findRouteNameFromNavigatorState } from '../../../util/general';
import { Text } from 'react-native';

const Stack = createStackNavigator();

const NavigationUnitTestFactory = ({ firstRoute, secondRoute }) => {
  const TestScreen = ({ route }) => {
    const routes = useNavigationState((state) => state.routes);

    const name = findRouteNameFromNavigatorState(routes);

    if (name !== route.params.screenName)
      throw new Error(
        'Error, react navigation api changed: https://reactnavigation.org/docs/navigation-prop/#dangerouslygetstate',
      );

    return <Text>{name} THIS SHOULD NOT HAVE CHANGED, take a deeper look</Text>;
  };

  const TestSubStack = () => (
    <Stack.Navigator initialRouteName="TestScreen">
      <Stack.Screen
        name="TestScreen3"
        component={TestScreen}
        initialParams={{ screenName: 'TestScreen3' }}
      />
    </Stack.Navigator>
  );

  const TestStack = () => (
    <Stack.Navigator initialRouteName={secondRoute || 'TestSubStack'}>
      <Stack.Screen name="TestSubStack" component={TestSubStack} />
      <Stack.Screen
        name="TestScreen2"
        component={TestScreen}
        initialParams={{ screenName: 'TestScreen2' }}
      />
    </Stack.Navigator>
  );

  const NavigationUnitTest = () => (
    <NavigationContainer>
      <Stack.Navigator initialRouteName={firstRoute || 'TestStack'}>
        <Stack.Screen name="TestStack" component={TestStack} />
        <Stack.Screen
          name="TestScreen1"
          component={TestScreen}
          initialParams={{ screenName: 'TestScreen1' }}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );

  return <NavigationUnitTest />;
};
export default NavigationUnitTestFactory;
