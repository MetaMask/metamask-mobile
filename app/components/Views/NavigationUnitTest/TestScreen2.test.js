import React from 'react';
import NavigationUnitTest from '.';
import { render } from 'enzyme';

describe('NavigationUnitTest', () => {
  it('should render correctly', () => {
    const wrapper = render(<NavigationUnitTest secondRoute={'TestScreen2'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
