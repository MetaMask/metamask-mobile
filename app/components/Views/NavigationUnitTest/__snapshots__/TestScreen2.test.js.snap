// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationUnitTest should render correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          collapsable={false}
          pointerEvents="box-none"
          style={
            {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          collapsable={false}
          pointerEvents="box-none"
          style={
            {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                collapsable={false}
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                TestStack
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <RNSScreenContainer
      onLayout={[Function]}
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        activityState={2}
        collapsable={false}
        forwardedRef={[Function]}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        pointerEvents="box-none"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            [
              {
                "overflow": undefined,
              },
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            {
              "close": {
                "animation": "spring",
                "config": {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": {
                "animation": "spring",
                "config": {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            collapsable={false}
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                {
                  "flex": 1,
                  "transform": [
                    {
                      "translateX": 0,
                    },
                    {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="none"
                style={
                  {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  [
                    {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    [
                      {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        pointerEvents="box-none"
                        style={
                          {
                            "zIndex": 1,
                          }
                        }
                      >
                        <View
                          accessibilityElementsHidden={false}
                          importantForAccessibility="auto"
                          onLayout={[Function]}
                          pointerEvents="box-none"
                          style={null}
                        >
                          <View
                            collapsable={false}
                            pointerEvents="box-none"
                            style={
                              {
                                "bottom": 0,
                                "left": 0,
                                "opacity": 1,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                                "zIndex": 0,
                              }
                            }
                          >
                            <View
                              collapsable={false}
                              style={
                                {
                                  "backgroundColor": "rgb(255, 255, 255)",
                                  "borderBottomColor": "rgb(216, 216, 216)",
                                  "flex": 1,
                                  "shadowColor": "rgb(216, 216, 216)",
                                  "shadowOffset": {
                                    "height": 0.5,
                                    "width": 0,
                                  },
                                  "shadowOpacity": 0.85,
                                  "shadowRadius": 0,
                                }
                              }
                            />
                          </View>
                          <View
                            collapsable={false}
                            pointerEvents="box-none"
                            style={
                              {
                                "height": 44,
                                "maxHeight": undefined,
                                "minHeight": undefined,
                                "opacity": undefined,
                                "transform": undefined,
                              }
                            }
                          >
                            <View
                              pointerEvents="none"
                              style={
                                {
                                  "height": 0,
                                }
                              }
                            />
                            <View
                              pointerEvents="box-none"
                              style={
                                {
                                  "alignItems": "center",
                                  "flex": 1,
                                  "flexDirection": "row",
                                  "justifyContent": "center",
                                }
                              }
                            >
                              <View
                                collapsable={false}
                                pointerEvents="box-none"
                                style={
                                  {
                                    "marginHorizontal": 16,
                                    "opacity": 1,
                                  }
                                }
                              >
                                <Text
                                  accessibilityRole="header"
                                  aria-level="1"
                                  collapsable={false}
                                  numberOfLines={1}
                                  onLayout={[Function]}
                                  style={
                                    {
                                      "color": "rgb(28, 28, 30)",
                                      "fontSize": 17,
                                      "fontWeight": "600",
                                    }
                                  }
                                >
                                  TestScreen2
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                      <RNSScreenContainer
                        onLayout={[Function]}
                        style={
                          {
                            "flex": 1,
                          }
                        }
                      >
                        <RNSScreen
                          activityState={2}
                          collapsable={false}
                          forwardedRef={[Function]}
                          gestureResponseDistance={
                            {
                              "bottom": -1,
                              "end": -1,
                              "start": -1,
                              "top": -1,
                            }
                          }
                          pointerEvents="box-none"
                          style={
                            {
                              "bottom": 0,
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            style={
                              {
                                "opacity": 1,
                              }
                            }
                          />
                          <View
                            accessibilityElementsHidden={false}
                            closing={false}
                            gestureVelocityImpact={0.3}
                            importantForAccessibility="auto"
                            onClose={[Function]}
                            onGestureBegin={[Function]}
                            onGestureCanceled={[Function]}
                            onGestureEnd={[Function]}
                            onOpen={[Function]}
                            onTransition={[Function]}
                            pointerEvents="box-none"
                            style={
                              [
                                {
                                  "overflow": undefined,
                                },
                                {
                                  "bottom": 0,
                                  "left": 0,
                                  "position": "absolute",
                                  "right": 0,
                                  "top": 0,
                                },
                              ]
                            }
                            transitionSpec={
                              {
                                "close": {
                                  "animation": "spring",
                                  "config": {
                                    "damping": 500,
                                    "mass": 3,
                                    "overshootClamping": true,
                                    "restDisplacementThreshold": 10,
                                    "restSpeedThreshold": 10,
                                    "stiffness": 1000,
                                  },
                                },
                                "open": {
                                  "animation": "spring",
                                  "config": {
                                    "damping": 500,
                                    "mass": 3,
                                    "overshootClamping": true,
                                    "restDisplacementThreshold": 10,
                                    "restSpeedThreshold": 10,
                                    "stiffness": 1000,
                                  },
                                },
                              }
                            }
                          >
                            <View
                              collapsable={false}
                              needsOffscreenAlphaCompositing={false}
                              pointerEvents="box-none"
                              style={
                                {
                                  "flex": 1,
                                }
                              }
                            >
                              <View
                                collapsable={false}
                                forwardedRef={[Function]}
                                onGestureHandlerEvent={[Function]}
                                onGestureHandlerStateChange={[Function]}
                                style={
                                  {
                                    "flex": 1,
                                    "transform": [
                                      {
                                        "translateX": 0,
                                      },
                                      {
                                        "translateX": 0,
                                      },
                                    ],
                                  }
                                }
                              >
                                <View
                                  collapsable={false}
                                  pointerEvents="none"
                                  style={
                                    {
                                      "backgroundColor": "rgb(242, 242, 242)",
                                      "bottom": 0,
                                      "left": 0,
                                      "position": "absolute",
                                      "shadowColor": "#000",
                                      "shadowOffset": {
                                        "height": 1,
                                        "width": -1,
                                      },
                                      "shadowOpacity": 0.3,
                                      "shadowRadius": 5,
                                      "top": 0,
                                      "width": 3,
                                    }
                                  }
                                />
                                <View
                                  style={
                                    [
                                      {
                                        "flex": 1,
                                        "overflow": "hidden",
                                      },
                                      [
                                        {
                                          "backgroundColor": "rgb(242, 242, 242)",
                                        },
                                        undefined,
                                      ],
                                    ]
                                  }
                                >
                                  <View
                                    style={
                                      {
                                        "flex": 1,
                                        "flexDirection": "column-reverse",
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        {
                                          "flex": 1,
                                        }
                                      }
                                    >
                                      <Text>
                                        TestScreen2
                                         THIS SHOULD NOT HAVE CHANGED, take a deeper look
                                      </Text>
                                    </View>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                        </RNSScreen>
                      </RNSScreenContainer>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenContainer>
  </RNCSafeAreaProvider>
</View>
`;
